editor/components/rich-text/index.js

```diff
@@ -411,9 +411,24 @@ export class RichText extends Component {
 		this.props.onChange( this.savedContent );
 	}
 
-	onCreateUndoLevel() {
-		// Always ensure the content is up-to-date.
-		this.onChange();
+	onCreateUndoLevel( event ) {
+		// TinyMCE fires a `change` event when the first letter in an instance
+		// is typed. This should not create a history record in Gutenberg.
+		// https://github.com/tinymce/tinymce/blob/4.7.11/src/core/main/ts/api/UndoManager.ts#L116-L125
+		// In other cases TinyMCE won't fire a `change` with at least a previous
+		// record present, so this is a reliable check.
+		// https://github.com/tinymce/tinymce/blob/4.7.11/src/core/main/ts/api/UndoManager.ts#L272-L275
+		if ( event && event.lastLevel === null ) {
+			return;
+		}
+
+		// Always ensure the content is up-to-date, but avoid dispatching an
+		// action if the original event is blur because the content will already
+		// be up-to-date.
+		if ( ! event || ! event.originalEvent || event.originalEvent.type !== 'blur' ) {
+			this.onChange();
```

Comment:
I tested and it works well for me, but just confirming one last thing. In which case the `onCreateUndoLevel` triggers while the `input` doesn't trigger? or is it related to the order of the events?

Reply:
E.g. making something bold will trigger change but not input, so generally for any "command" in TinyMCE. Maybe we can also do the opposite here and call onChange for any command and other events that need a change.

Reply:
I see, I think it sounds good to me 👍

Reply:
@youknowriad I updated the comment so it's clearer now.